### PR TITLE
chore(compass-import-export): Convert import-export to a global plugin instead of a scoped one COMPASS-5537 COMPASS-4911 COMPASS-5143 

### PR DESCRIPTION
**Pull request by gribnoysup**
Another prerequisite patch for the upcoming work on run and export aggregations feature. This PR decouples `compass-import-export` plugin from the whole collection workspace. Instead of being created for every collection tab separately and both listening to local `open` events and updates to the local query bar state, both import and export are now singleton plugins that can be activated by global app registry events that will provide all the initial information needed for the import / export work.

This means that when we start working on aggregation export feature, we don't need to worry about awkward state connections between parent plugin and the scoped one, we would just need to adjust the app registry event to support aggregations and then all the other work for the actual export can be done completely in isolation.

As a bonus this also fixes https://jira.mongodb.org/browse/COMPASS-4911 and  https://jira.mongodb.org/browse/COMPASS-5143

### CODE REVIEW

File: packages/compass-import-export/src/modules/export.spec.js [hunk 1]
```diff
@@ -1,68 +1,69 @@
...
-        }
 
-        store = configureExportStore({
-          localAppRegistry: localAppRegistry,
-          globalAppRegistry: globalAppRegistry,
-          namespace: TEST_COLLECTION_NAME,
-          dataProvider: {
-            error: function(err) { throw err; },
-            dataProvider: dataService
+        await createCollection(TEST_COLLECTION_NAME);
```
Comment (Anemy) [inline packages/compass-import-export/src/modules/export.spec.js:35]: Nice, I like we're not silently ignoring the errors now.

File: packages/compass-crud/src/stores/crud-store.js [hunk 2]
```diff
@@ -717,16 +717,22 @@ const configureStore = (options = {}) => {
...
     /**
      * Open an export file dialog from compass-import-export-plugin.
      * Emits a global app registry event the plugin listens to.
      */
     openExportFileDialog() {
-      // Pass the doc count to the export modal so we can avoid re-counting.
-      this.localAppRegistry.emit('open-export', this.state.count);
+      // Only three query fields that export modal will handle
+      const { filter, limit, skip } = this.state.query;
+      this.globalAppRegistry.emit('open-export', {
```
Comment (Anemy) [inline packages/compass-crud/src/stores/crud-store.js:730]: Nicee looks like this fixes https://jira.mongodb.org/browse/COMPASS-5143 too? I pulled it into sprint if that's cool.

File: packages/compass-import-export/src/modules/export.js [hunk 3]
```diff
@@ -513,14 +510,6 @@ export const startExport = () => {
         fileName: exportData.fileName,
       });
       dispatch(onFinished(numDocsToExport));
-      dispatch(
-        appRegistryEmit(
-          'export-finished',
```
Comment (Anemy) [inline packages/compass-import-export/src/modules/export.js:518]: Was nothing listening to this event, only the global one?

**Review by Anemy — APPROVED** [review]
Nice - probably a slight performance increase too when a new tab is created.
I like how we're making it more straight forward here.

File: packages/compass-crud/src/stores/crud-store.js (hunk 2 above)
Comment (gribnoysup) [reply packages/compass-crud/src/stores/crud-store.js:730]: Haha, I didn't even know this was a bug before, yeah, seems to be fixed, thanks for spotting

File: packages/compass-import-export/src/modules/export.js (hunk 3 above)
Comment (gribnoysup) [reply packages/compass-import-export/src/modules/export.js:518]: Yep, nothing in the app, only old analytics seems to be listening to this one

Comment (gribnoysup) [reply packages/compass-import-export/src/modules/export.js:518]: After this comment I double-checked other events that were affected by this change and did a clean-up for import-finished too in 0704881430f6f74eb9247501ddd0141a39b4a40a

**Review by Anemy — APPROVED** [review]
One less global `refresh-data` and more localized refreshes 🥳

**Pull request merged**